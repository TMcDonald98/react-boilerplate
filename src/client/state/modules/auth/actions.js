
import { constants } from 'app';

export const setApiKey = apiKey => ({
  type: constants.setApiKey,
  payload: apiKey
});
