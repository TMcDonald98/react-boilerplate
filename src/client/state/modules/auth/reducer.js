import { constants } from 'app';

const defaultState = {
  apiKey: null
};

const auth = (state = defaultState, action) => {
  switch(action.type) {
    case constants.setApiKey:
      return {
        ...state,
        apiKey: action.payload
      };
    default:
      return state;
  }
};

export { auth, defaultState };
