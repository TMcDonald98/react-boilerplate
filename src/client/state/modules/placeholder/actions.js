
import { constants } from 'app';

export const setPlaceholderData = placeholderData => ({
  type: constants.setPlaceholderData,
  payload: placeholderData
});

export const clearPlaceholderData = () => ({
  type: constants.clearPlaceholderData
});
