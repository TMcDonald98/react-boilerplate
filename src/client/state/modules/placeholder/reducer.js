import { constants } from 'app';

const defaultState = {
  data: null
};

const placeholder = (state = defaultState, action) => {
  switch(action.type) {
    case constants.setPlaceholderData:
      return {
        ...state,
        data: action.payload
      };
    case constants.clearPlaceholderData:
      return {
        ...state,
        data: defaultState.data
      };
    default:
      return state;
  }
};

export { placeholder, defaultState };
