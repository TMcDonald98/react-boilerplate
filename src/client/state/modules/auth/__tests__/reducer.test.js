import { auth } from '../reducer';
import * as actions from '../actions';

describe('auth/reducer', () => {
  describe('default', () => {
    it('should trigger the default case', () => {
      const result = auth(undefined, { type: '' });

      expect(result).toMatchSnapshot('default');
    });
  });
  describe('setApiKey', () => {
    it('should trigger the setApiKey case', () => {
      const result = auth(undefined, actions.setApiKey('success'));

      expect(result).toMatchSnapshot('setApiKey');
    });
  });
});