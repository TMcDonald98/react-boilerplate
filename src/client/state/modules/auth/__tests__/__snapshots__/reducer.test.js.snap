// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth/reducer default should trigger the default case: default 1`] = `
Object {
  "apiKey": null,
}
`;

exports[`auth/reducer setApiKey should trigger the setApiKey case: setApiKey 1`] = `
Object {
  "apiKey": "success",
}
`;
